### PR TITLE
Implement CharSections.dbc and Serverside Checks.

### PR DESCRIPTION
**Pull request by ariel-**
Implements checks on player Load, Enum, Barber Shop and Customization against packet modification (physical appearance)

Need feedback on this one, tested as much as I could, eliminating false possitives, now I'm in doubt if it let's go true ones.

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -1981,13 +1986,32 @@ bool Player::BuildEnumData(PreparedQueryResult result, WorldPacket* data)
...
+        TC_LOG_ERROR("entities.player.loading", "Player %u has wrong Appearance values (Hair/Skin/Color), enumerating defaults and forcing recustomize", guid); 
+        if (!(atLoginFlags & AT_LOGIN_CUSTOMIZE))
+        {
+            PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_UPD_ADD_AT_LOGIN_FLAG);
+            stmt->setUInt16(0, uint16(AT_LOGIN_CUSTOMIZE));
+            stmt->setUInt32(1, guid);
+            CharacterDatabase.Execute(stmt);
+            atLoginFlags |= AT_LOGIN_CUSTOMIZE;
+        }
+        *data << uint8(0) << uint8(0) << uint8(0) << uint8(0) << uint8(0);
```
Comment (Shauren) [inline src/server/game/Entities/Player/Player.cpp]: You don't really need to override data sent in enum packet, just setting customize flag is enough (only character owner can see it)

File: src/server/scripts/Northrend/Nexus/Nexus/boss_magus_telestra.cpp [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 #include "ScriptMgr.h"
 #include "ScriptedCreature.h"
+#include "GameEventMgr.h"
```
Comment (Frytiks) [inline src/server/scripts/Northrend/Nexus/Nexus/boss_magus_telestra.cpp]: For what is this ?

Comment (ariel-) [inline src/server/scripts/Northrend/Nexus/Nexus/boss_magus_telestra.cpp]: Not related to the PR, but instead compiled again with no PCH and added missing include so it Builds correctly.

**Comment by ariel-** [comment]
Ready.

**Comment by DDuarte** [comment]
What is necessary to update this to `6.x`?

**Comment by ariel-** [comment]
```c++
struct CharSectionsRec {
  uint32_t m_ID;
  uint32_t m_raceID;
  uint32_t m_sexID;
  uint32_t m_baseSection;
  stringref m_TextureName[3];
  uint32_t m_flags;
  uint32_t m_variationIndex;
  uint32_t m_colorIndex;
};
```

DBC structure changed, ATM I'm only in the 3.3.5 branch; dunno but I guess client-server communication changed too.

The loading part should go smooth tho, as its using racemask.

**Comment by danlapps** [comment]
Is this ready for merge in 335 ?

**Comment by Carbenium** [comment]
:shipit:

**Pull request merged**